linkerd/app/inbound/src/lib.rs

```diff
@@ -275,18 +283,28 @@ impl Config {
             .into_inner()
     }
 
-    pub async fn build_server<C, H, S>(
+    pub fn build_server<C, H, S>(
         self,
-        listen_addr: std::net::SocketAddr,
-        listen: impl Stream<Item = std::io::Result<listen::Connection>> + Send + 'static,
         prevent_loop: impl Into<PreventLoop>,
         tcp_connect: C,
         http_router: H,
         local_identity: tls::Conditional<identity::Local>,
         metrics: ProxyMetrics,
         span_sink: Option<mpsc::Sender<oc::Span>>,
         drain: drain::Watch,
-    ) -> Result<(), Error>
+    ) -> impl tower::Service<
+        listen::Addrs,
+        Error = impl Into<Error>,
+        Future = impl Send + 'static,
+        Response = impl tower::Service<
+            tokio::net::TcpStream,
+            Response = (),
+            Error = impl Into<Error>,
+            Future = impl Send + 'static,
+        > + Send
+                       + 'static,
+    > + Send
+           + 'static
```

Comment:
IIRC we weren't returning these because there was an impact on build times, is that still the case?

Reply:
It doesn't appear to be... builds are basically in line with where they are on main (which isn't so speedy, either ;)

Reply:
Eh, it does seem that this, on top of the prior changes, makes integration compilation pretty slow. I'm going to merge it but I think this all in general puts more pressure on us to figure out how to avoid integration tests being soooo costly.